docs(swinv2): Update SwinV2 model card to new standard format

BryanBradfo opened the pull request:
# What does this PR do?

This PR updates the Swin Transformer V2 model card (`swin_transformer_v2.md`) to align with the new standardized format.

-   Follows the template provided in [Community contributions] Model cards #36979.
-   Includes working Pipeline and AutoModel examples (tested locally).
-   Removed the transformers-cli example (consistent with other recent model card updates).

Relates to #36979

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu

BryanBradfo committed:
docs(swinv2): Update SwinV2 model card to new standard format

stevhliu commented on docs/source/en/model_doc/swinv2.md:
```diff
@@ -14,37 +14,74 @@ rendered properly in your Markdown viewer.
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-</div>
+[Swin Transformer V2](https://huggingface.co/papers/2111.09883) builds upon the original Swin Transformer, focusing on scaling up model capacity and training resolution effectively. It introduces techniques like residual-post-norm combined with cosine attention for improved training stability, and a log-spaced continuous position bias to better handle varying image resolutions between pre-trainin ...
```
```suggestion
[Swin Transformer V2](https://huggingface.co/papers/2111.09883) is a 3B parameter model that focuses on how to scale a vision model to billions of parameters. It introduces techniques like residual-post-norm combined with cosine attention for improved training stability, log-spaced continuous position bias to better handle varying image resolutions between pre-training and fine-tuning, and a new pre-training method (SimMIM) to reduce the need for large amounts of labeled data. These improvements enable efficiently training very large models (up to 3 billion parameters) capable of processing high-resolution images.
```

stevhliu commented on docs/source/en/model_doc/swinv2.md:
```diff
@@ -14,37 +14,74 @@ rendered properly in your Markdown viewer.
+You can find official Swin Transformer V2 checkpoints under the [Microsoft](https://huggingface.co/microsoft?search_models=swinv2) organization.
+
+> [!TIP]
+> Click on the Swin Transformer V2 models in the right sidebar for more examples of how to apply Swin Transformer V2 to image classification tasks.
```
```suggestion
> Click on the Swin Transformer V2 models in the right sidebar for more examples of how to apply Swin Transformer V2 to vision tasks.
```

stevhliu reviewed (approved):
Super clean implementation! 👏

BryanBradfo committed:
docs(swinv2): Apply review suggestions

Incorporates feedback from @stevhliu to:
- Enhance the introductory paragraph with more details about scaling and SimMIM.
- Generalize the tip from "image classification tasks" to "vision tasks".

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>
